pkg/cloud/azure/services/availabilitysets/availabilitysets.go

```diff
@@ -21,6 +24,42 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 	if !ok {
 		return errors.New("invalid availability set specification")
 	}
+	// Get the MaximumPlatformFaultDomainCount
+	skuService := resourceskus.NewService(s.Scope)
+	skuSpec := resourceskus.Spec{
+		Name:         string(compute.AvailabilitySetSkuTypesAligned),
+		ResourceType: resourceskus.AvailabilitySets,
+	}
+
+	skuI, err := skuService.Get(ctx, skuSpec)
+	if err != nil {
+		if errors.Is(err, resourceskus.ErrResourceNotFound) {
+			return machinecontroller.InvalidMachineConfiguration("failed to obtain availability set information for Aligned '%s' from Azure: %s", skuSpec.Name, err)
+		} else {
+			return fmt.Errorf("failed to obtain aligned information: %w", err)
+		}
+	}
+
+	sku := skuI.(resourceskus.SKU)
+
+	fmt.Printf("Capabilities: %+v\n", sku.Capabilities)
```

Comment:
this should be a log call, perhaps `klog.Infof("Capabilities: %+v\n", sku.Capabilities)`